src/useMatches.tsx

```diff
@@ -74,7 +74,29 @@ export function useMatches() {
     return getDeepResults(rootResults);
   }, [getDeepResults, rootResults, emptySearch]);
 
-  const matches = useInternalMatches(filtered, search);
+  const fuseOptions = {
+    keys: [{
+      name: "name",
+      weight: 0.5
+    },
+    {
+      name: "keywords",
+      getFn: (item) => item.keywords.split(","), // make keyword an array. so fuse can look through words individually
+      weight: 0.5
+    },
+      "subtitle"],
+    includeScore: true,
+    includeMatches: true,
+    threshold: 0.2,
+    minMatchCharLength: 1,
+    tokenize: (str) => {
+      // Example: Preserve hyphens and special characters as separate tokens
+      return str.split(/[\s\-,.!()]+/).filter(Boolean);
+    },
+  };
+  const fuse = new Fuse(filtered, fuseOptions);
```

Comment:
Don't you want to move this inside the `useMemo` and only create a new fuse instance when `filtered` changes? It's essentially rebuilding the search index on every render

Reply:
Yah https://github.com/timc1/kbar/pull/329